FWCore/ParameterSet/test/test_catch2_ps.cc

```diff
@@ -0,0 +1,875 @@
+/*
+ */
+#define CATCH_CONFIG_MAIN
+#include "catch.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <limits>
+#include <string>
+#include <cassert>
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/types.h"
+#include "FWCore/Utilities/interface/EDMException.h"
+#include "FWCore/Utilities/interface/Algorithms.h"
+#include "FWCore/Utilities/interface/Digest.h"
+namespace edm {
+  bool operator!=(const edm::EventRange& iLHS, const edm::EventRange& iRHS) {
+    return ((iLHS.startEventID() != iRHS.startEventID()) or (iLHS.endEventID() != iRHS.endEventID()));
+  }
+}  // namespace edm
+namespace {
+  void do_compare(const edm::EventRange& iLHS, const edm::EventRange& iRHS) {
+    CHECK(iLHS.startEventID() == iRHS.startEventID());
+    CHECK(iLHS.endEventID() == iRHS.endEventID());
+  }
+
+  template <class T>
+  void do_compare(const T& iLHS, const T& iRHS) {
+    CHECK(iLHS == iRHS);
+  }
+
+  template <class T, class A>
+  void do_compare(const std::vector<T, A>& iLHS, const std::vector<T, A>& iRHS) {
+    REQUIRE_THAT(iLHS, Catch::Matchers::Equals(iRHS));
+  }
+
+  template <class T>
+  void trackedTestbody(const T& value) {
+    edm::ParameterSet p1;
+    p1.template addParameter<T>("x", value);
+    p1.registerIt();
+    do_compare(p1.template getParameter<T>("x"), value);
+    std::string p1_encoded = p1.toString();
+    edm::ParameterSet p2(p1_encoded);
+    REQUIRE(p1 == p2);
+    do_compare(p2.template getParameter<T>("x"), value);
+  }
+
+  template <class T>
+  void untrackedTestbody(const T& value) {
+    edm::ParameterSet p;
+    p.template addUntrackedParameter<T>("x", value);
+    do_compare(p.template getUntrackedParameter<T>("x"), value);
+
+    REQUIRE_THROWS_AS(p.template getUntrackedParameter<T>("does not exist"), cms::Exception);
+  }
+
+  template <class T>
+  void testbody(T value) {
+    trackedTestbody<T>(value);
+    untrackedTestbody<T>(value);
+  }
+
+  template <class T>
+  void test_for_name() {
+    edm::ParameterSet preal;
+    edm::ParameterSet const& ps = preal;
+    // Use 'ps' to make sure we're only getting 'const' access;
+    // use 'preal' when we need to modify the underlying ParameterSet.
+
+    std::vector<std::string> names = ps.getParameterNames();
+    REQUIRE(names.empty());
+
+    // The following causes failure, because of an apparent GCC bug in comparing bools!
+    //T value;
```

Comment:
I wonder if this comment still holds

Reply:
probably not (again, just a left over from the cppunit conversion)

Reply:
removed